contracts/contracts/Job.sol

```diff
@@ -233,148 +262,233 @@ contract Job {
+        sendJobPayout(jobs[jobId].token, payoutAmount, daoTakeAmount, jobs[jobId].supplier);
 
         emit JobDisputeResolved(jobId, States.FinalDisputeResolvedForSupplier);
     }
 
-    function resolveDisputeForEngineer(uint jobId) public requiresJobState(jobId, States.Disputed) requiresResolver() {
+    function resolveDisputeForEngineer(uint256 jobId) public onlyResolver requiresJobState(jobId, States.Disputed) {
         jobs[jobId].state = States.FinalDisputeResolvedForEngineer;
 
-        (uint payoutAmount, uint daoTakeAmount) = calculateFullDisputeResolutionPayout(jobs[jobId].bounty, jobs[jobId].deposit);
-        sendJobPayout(payoutAmount, daoTakeAmount, address(jobs[jobId].engineer));
+        (uint256 payoutAmount, uint256 daoTakeAmount) = calculateFullDisputeResolutionPayout(
+            jobs[jobId].bounty,
+            jobs[jobId].deposit
+        );
+        sendJobPayout(jobs[jobId].token, payoutAmount, daoTakeAmount, jobs[jobId].engineer);
 
         emit JobDisputeResolved(jobId, States.FinalDisputeResolvedForEngineer);
     }
 
-    function resolveDisputeWithCustomSplit(uint jobId, uint engineerAmountPct) public requiresJobState(jobId, States.Disputed) requiresResolver() {
+    function resolveDisputeWithCustomSplit(uint256 jobId, uint256 engineerAmountPct)
+    public
+    onlyResolver
+    requiresJobState(jobId, States.Disputed)
+    {
         require(engineerAmountPct >= MINIMUM_SPLIT_CHUNK_PERCENTAGE, "Percentage too low");
         require(engineerAmountPct <= BASE_PERCENTAGE - MINIMUM_SPLIT_CHUNK_PERCENTAGE, "Percentage too high");
 
         jobs[jobId].state = States.FinalDisputeResolvedWithSplit;
 
-        (uint supplierPayoutAmount, uint engineerPayoutAmount, uint daoTakeAmount) = calculateSplitDisputeResolutionPayout(jobs[jobId].bounty, jobs[jobId].deposit, engineerAmountPct);
-        sendSplitJobPayout(jobId, supplierPayoutAmount, engineerPayoutAmount, daoTakeAmount);
+        JobData memory job = jobs[jobId];
+
+        (
+        uint256 supplierPayoutAmount,
+        uint256 engineerPayoutAmount,
+        uint256 daoTakeAmount
+        ) = calculateSplitDisputeResolutionPayout(job.bounty, job.deposit, engineerAmountPct);
+        sendSplitJobPayout(job, supplierPayoutAmount, engineerPayoutAmount, daoTakeAmount);
 
         emit JobDisputeResolved(jobId, States.FinalDisputeResolvedWithSplit);
     }
 
+    // TODO: Do we need any convenient getters ?
+
     /****************************
      * DAO Management Functions *
      ****************************/
 
-    function withdrawDaoFunds(address recipient, uint amount) public requiresFundManager() {
-        require(amount <= daoFunds, "Insufficient funds");
+    // TODO: Add a safety function to make sure that funds can never be locked in the contract.
+    // TODO: "initiateWithdraw" which starts a 24h timer after which "withdraw(jobId)" can be called by onlyOwner()
+
+    // TODO: what if someone sends a token by mistake to this contract ?
```

Comment:
I thought about this a bit too.  By allowing arbitrary withdrawals of funds, we create a larger surface area for an attack.  If someone gets the contract private key(s) they could drain both the daoFunds and the daoEscrow if we have an emergency withdrawal function.

I think we should be conservative and just not allow arbitrary ERC20 withdrawals even by the owner.

We might want to add a function to withdraw ETH, though...